packages/components/src/dropdown-menu-v2-ariakit/index.tsx

```diff
@@ -248,6 +248,29 @@ const UnconnectedDropdownMenu = (
 		);
 	}
 
+	const hideOnEscape = useCallback(
+		( event: React.KeyboardEvent< Element > ) => {
+			// Pressing Escape can cause unexpected consequences (ie. exiting
+			// full screen mode on MacOs, close parent modals...).
+			event.preventDefault();
+			event.stopPropagation();
```

Comment:
Is there any particular reason to stop propagation?

Reply:
Without it, I thought that, when a DropdownMenu is rendered inside a Modal, pressing Escape would close the DropdownMenu and the parent modal.

That was my impression because, when talking to Diego, I seem to remember him suggesting [this change](https://github.com/WordPress/gutenberg/blob/1b5f4ef781b83d27918a21865302b914d2eb4281/packages/components/src/dropdown-menu-v2-ariakit/stories/index.story.tsx#L497-L500) specifically for when a DropdownMenu is rendered inside a Modal.

But I just tested without it, and things seem to work well.